src/pymovements/events/base.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module holds all the main Event classes used for event detection.
+"""
+
+
+class Event:
+    """
+    Base Event class.
+
+    Parameters
+    ----------
+    name: str
+        Name of event.
+    onset: int
+        Starting index of event (included).
+    offset: int
+        Ending index of event (excluded).
+
+    Attributes
+    ----------
+    name: str
+        Name of event.
+    onset: int
+        Starting index of event (included).
+    offset: int
+        Ending index of event (excluded).
+
+    """
+    def __init__(self, name: str, onset: int, offset: int):
+        self.name = name
+        self.onset = onset
+        self.offset = offset
+
+    @property
+    def duration(self) -> int:
+        """
+        Get sample duration of event.
+
+        Returns
+        -------
+        int
+            duration in samples.
+        """
+        return self.offset - self.onset
```

Comment:
since time here is samples -- when do we plan to add ms/sec

Reply:
You are totally right that this will have to be extended somehow to increase usability.

I addressed this issue in #50